components/style/properties.mako.rs

```diff
@@ -258,6 +258,75 @@ pub mod longhands {
     </%def>
 
 
+    <%def name="integer_type_or_auto(name, type, initial_value='SpecifiedValue::Auto',
+                                     condition='true', experimental='False')">
+        <%self:longhand name="${name}", experimental="${experimental}">
+            use values::computed::Context;
+            use cssparser::ToCss;
+            use std::fmt;
+
+            #[derive(Clone, Copy, PartialEq)]
+            pub enum SpecifiedValue {
+                Auto,
+                Specified(${type}),
+            }
+
+            impl ToCss for SpecifiedValue {
+                fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+                    match *self {
+                        SpecifiedValue::Auto => dest.write_str("auto"),
+                        SpecifiedValue::Specified(count) => write!(dest, "{}", count),
+                    }
+                }
+            }
+
+            pub mod computed_value {
+                #[derive(Clone, PartialEq, HeapSizeOf)]
+                pub struct T(pub Option<${type}>);
+            }
+
+            impl ToCss for computed_value::T {
+                fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+                    match self.0 {
+                        None => dest.write_str("auto"),
+                        Some(count) => write!(dest, "{}", count),
+                    }
+                }
+            }
+
+            #[inline]
+            pub fn get_initial_value() -> computed_value::T {
+                computed_value::T(None)
+            }
+
+            impl ToComputedValue for SpecifiedValue {
+                type ComputedValue = computed_value::T;
+
+                #[inline]
+                fn to_computed_value(&self, _context: &Context) -> computed_value::T {
+                    match *self {
+                        SpecifiedValue::Auto => computed_value::T(None),
+                        SpecifiedValue::Specified(count) =>
+                            computed_value::T(Some(count))
+                    }
+                }
+            }
+
+            pub fn parse(_context: &ParserContext, input: &mut Parser) -> Result<SpecifiedValue, ()> {
+                if input.try(|input| input.expect_ident_matching("auto")).is_ok() {
+                    Ok(SpecifiedValue::Auto)
+                } else {
+                    let value = try!(input.expect_integer());
+                    if !(${condition}) {
+                        return Err(())
+                    }
+                    Ok(SpecifiedValue::Specified(value as ${type}))
```

Comment:
That would make sense, but I don't think redundant / trivial casts ever do warn. At least, there aren't any warnings when I build this file using `./mach build`, and http://is.gd/9aTZQ1 doesn't show any warnings when compiling.